lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1731,9 +1752,21 @@ optimizeBridgedSwiftToObjCCast(SILInstruction *Inst,
       }
       break;
     case ParameterConvention::Direct_Unowned:
+      assert(!AddressOnlyType &&
+             "AddressOnlyType with Direct_Unowned is not supported");
       break;
-    case ParameterConvention::Indirect_In:
     case ParameterConvention::Indirect_In_Guaranteed:
+      // Source as-is, we don't need to copy it due to guarantee
+      break;
+    case ParameterConvention::Indirect_In: {
+      assert(AddressOnlyType &&
+             "Expected an AddressOnlyType with Indirect_In Convention");
```

Comment:
Changed it as advised + changed your new isIndirect() bailout since we *do* support some indirect conventions now